template_handler/cdk.py

```diff
@@ -0,0 +1,199 @@
+            f'npm list {package}',
+            shell=True,
+            capture_output=True
+        )
+        self.logger.debug(f"Workspace NPM package '{package}' exists: {not bool(workspace_result.returncode)}")
+
+        if workspace_result.returncode == 0:
+            package_exists = True
+        else:
+            global_result = subprocess.run(
+                f'npm --global list {package}',
+                shell=True,
+                capture_output=True
+            )
+            self.logger.debug(f"Global NPM package '{package}' exists: {not bool(global_result.returncode)}")
+
+            if global_result.returncode == 0:
+                package_exists = True
+
+        return package_exists
+
+    def _check_prerequisites(self) -> None:
+        """
+        Checks the command and Node package requirements for the handler
+        """
+
+        # Check Command Prerequisites
+        cmd_prerequisites = [
+            'node',
+            'npx'
+        ]
+        for cmd_prerequisite in cmd_prerequisites:
+            if not self._cmd_exists(cmd_prerequisite):
+                raise exceptions.SceptreException(f"Command prerequisite '{cmd_prerequisite}' not found")
+
+        # Check Node Package Prerequisites
+        node_prerequisites = [
+            'cdk-assets'
+        ]
+        for node_prerequisite in node_prerequisites:
+            if not self._node_package_exists(node_prerequisite):
+                raise exceptions.SceptreException(f"Node Package prerequisite '{node_prerequisite}' not found")
+
+    def handle(self) -> str:
+        """
+        Main Sceptre CDK Handler function
+
+        Returns
+        -------
+        str|bytes
+            CDK synthesised CloudFormation template
+        """
+
+        self._check_prerequisites()
+
+        # Import CDK Python template module
+        template_path = posixpath.join('templates', self.cdk_template_path)
+        self.logger.debug(f'Importing CDK Python template module {template_path}')
+        template_module_name = 'cdk_template'
+        loader = importlib.machinery.SourceFileLoader(template_module_name, template_path)
+        spec = importlib.util.spec_from_loader(template_module_name, loader)
+        template_module = importlib.util.module_from_spec(spec)
+        loader.exec_module(template_module)
+
+        # CDK Synthesize App
+        self.logger.debug('CDK synthesing CdkStack Class')
+        app = aws_cdk.App()
+        stack_name = 'CDKStack'
+        template_module.CdkStack(app, stack_name, self.sceptre_user_data)
+        app_synth = app.synth()
+
+        # Publish CDK Assets
+        asset_artifacts = None
+
+        for artifacts in app_synth.artifacts:
+            if isinstance(artifacts, aws_cdk.cx_api.AssetManifestArtifact):
+                asset_artifacts = artifacts
+                break
+        if asset_artifacts is None:
+            raise exceptions.SceptreException('Asset manifest artifact not found')
```

Comment:
Will there ALWAYS be a manifest? Even if there are no assets to publish? For MOST CDK constructs, there will be no assets. It's just a few select ones that actually have assets.